ironic/drivers/modules/drac/inspect.py

```diff
@@ -35,16 +42,168 @@
+
+        redfish_ip = task.node.driver_info.get('redfish_address')
+        redfish_username = task.node.driver_info.get('redfish_username')
+        redfish_password = task.node.driver_info.get('redfish_password')
+
+        if system.boot.mode == 'uefi':
+            for param, nic in zip(pxe_params, pxe_nics):
+                if system.bios.attributes[param] == 'Enabled':
+                    nic_id = system.bios.attributes[nic]
+                    # Get MAC address of the given nic_id
+                    mac_address = self._get_mac_address(
+                        [nic_id], redfish_ip, redfish_username,
+                        redfish_password)
+                    pxe_dev_nics.append(mac_address)
+
+        elif system.boot.mode == 'bios':
+            url = '%s%s' % (redfish_ip, _JOB_ID_URL)
+            job_response = requests.post(url, data=json.dumps(
+                _PAYLOAD), headers=_HEADERS, verify=False, auth=(
+                redfish_username, redfish_password))
+
+            if job_response.status_code != _JOB_RESPONSE_CODE:
+                error = (_('An error occurred when attempting to export '
+                           'the system configuration. Status code: %(code), '
+                           'Error details: %(err)'),
+                         {'code': job_response.status_code,
+                          'err': job_response.__dict__})
+                LOG.error(error)
+                raise exception.HardwareInspectionFailure(error=error)
+
+            job_id = job_response.__dict__['headers']['Location']
+            try:
+                job_id = re.search('JID_.+', job_id).group()
+            except drac_exceptions.BaseClientException as exc:
+                LOG.error('Unable to find the job id in response: %(err)',
+                          {'err': job_response.__dict__['_content']})
+                raise exception.HardwareInspectionFailure(error=exc)
+
+            start_time = datetime.now()
+
+            while True:
+                current_time = (datetime.now() - start_time)
+                xml_data = requests.get('%s%s%s' % (
+                    redfish_ip, _XML_DATA_URL, job_id), headers=_HEADERS,
+                    auth=(redfish_username, redfish_password), verify=False)
+
+                if _SYSTEM_CONFIG_TAG in str(xml_data.__dict__):
+                    xml_string = xml_data.__dict__.get('_content')
+                    root = ET.fromstring(xml_string)
+                    for child in root:
+                        nic_id = [child.attrib.get('FQDD') for ch in child
+                                  if ch.text == 'PXE']
+                        if nic_id:
+                            # Get MAC address of the given nic_id
+                            mac_address = self._get_mac_address(
+                                nic_id, redfish_ip,
+                                redfish_username, redfish_password)
+                            pxe_dev_nics.append(mac_address)
```

Comment:
The above lines depends on the firmware returning the MAC address in upper case.  Let's make it mac_address.upper() so that it won't matter if the firmware returns it in upper or lower case.